### PR TITLE
[ADS-6051] chore: upgrade to mdx v2

### LINKED ISSUE #1223: upgrade @mdx
## Types of issue

- [ ] Bug (High impact or critical to a production build)
- [ ] Bug (Low impact or not working as expected)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Something else (idea, build tool, documentation improvement)

## Description

upgrade:
- @mdx-js/loader  
- @mdx-js/react 

ensure nothing breaks in documentation app

### PR DESCRIPTION
**Pull request by glenkurniawan-adslot**
## Description

- upgrade mdx to v2
- migrate dev scripts to esm except jest config related

Fixes #1223 

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Review by pphminions — APPROVED** [review]
LGTM 🖌️

**Review by bchew — DISMISSED** [review]

File: www/containers/CodeBlock.jsx [hunk 1]
```diff
@@ -27,6 +27,7 @@ export default ({ children, className, live }) => {
       </LiveProvider>
     );
   }
+  console.log('not live');
```
Comment (glenkurniawan-adslot) [inline www/containers/CodeBlock.jsx:30]: remove this console.log... :)

**Review by bchew — APPROVED** [review]
🏆

File: www/containers/CodeBlock.jsx (hunk 1 above)
Comment (knilink) [reply www/containers/CodeBlock.jsx]: fixed :+1:

File: config/webpack.config.dev.js [hunk 2]
```diff
@@ -46,10 +47,17 @@ module.exports = webpackMerge(commonConfig, {
           {
             loader: '@mdx-js/loader',
             options: {
-              remarkPlugins: [emoji],
+              providerImportSource: '@mdx-js/react',
+              remarkPlugins: [emoji, remarkMdxCodeMeta],
```
Comment (renrizzolo) [inline config/webpack.config.dev.js:51]: this needs to be adde to `webpack.config.prod`

Comment (knilink) [reply config/webpack.config.dev.js:51]: fixed :+1:

**Review by renrizzolo — APPROVED** [review]

**Review by bchew — APPROVED** [review]

**Review by xiaofan2406 — APPROVED** [review]

**Review by xiaofan2406 — APPROVED** [review]

**Pull request merged**